### PR TITLE
Android: Support "Start On Boot" 

### PR DESCRIPTION
**Pull request by strseb**
Adds Support for start on boot.

### CODE REVIEW

File: src/platforms/android/androidvpnactivity.cpp [hunk 1]
```diff
@@ -142,3 +149,9 @@ void AndroidVPNActivity::onServiceDisconnected(JNIEnv* env, jobject thiz) {
   logger.debug() << "service disconnected";
   emit AndroidVPNActivity::instance()->serviceDisconnected();
 }
+void AndroidVPNActivity::startAtBootChanged(const bool& startAtBoot) {
+  QJsonObject args;
+  args["startOnBoot"] = startAtBoot;
+  QJsonDocument doc(args);
+  sendToService(ServiceAction::ACTION_SET_START_ON_BOOT, doc.toJson());
+}
```
Comment (brizental) [inline src/platforms/android/androidvpnactivity.cpp:157]: nit: New line

File: src/featureslistcallback.h [hunk 2]
```diff
@@ -150,7 +150,7 @@ bool FeatureCallback_splitTunnel() {
 
 bool FeatureCallback_startOnBoot() {
 #if defined(MVPN_LINUX) || defined(MVPN_MACOS) || defined(MVPN_WINDOWS) || \
-    defined(MVPN_DUMMY) || defined(MVPN_WASM)
+    defined(MVPN_DUMMY) || defined(MVPN_WASM) || defined(MVPN_ANDROID)
```
Comment (brizental) [inline src/featureslistcallback.h:153]: nit: At this point should we just check if it's **not** iOS instead?

File: android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/BootReceiver.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    override fun onReceive(context: Context, arg1: Intent) {
+        Log.init(context)
+        if (!Prefs.get(context).getBoolean(START_ON_BOOT, false)) {
+            Log.i(TAG, "This device did not enable start on boot - exit")
+            return
+        }
+        Log.i(TAG, "This device did enable start on boot - try to start")
+        val intent = Intent(context, VPNService::class.java)
+        intent.putExtra("startOnBoot", true)
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
```
Comment (brizental) [inline android/daemon/src/main/java/org/mozilla/firefox/vpn/daemon/BootReceiver.kt:24]: Just to see if I get it. We are checking if the current Android SDK version is greater than Android SDK version `Oreo`? Ha... That took me a while, why does 0 look so much like O???????? Anyways, would you mind adding a comment here on why we need to check this?

**Review by brizental — APPROVED** [review]
Not sure I am the right reviewer here but... r+wc :trophy

**Review by bakulf — APPROVED** [review]

**Pull request merged**